packages/babel-helpers/src/helpers.js

```diff
@@ -73,11 +73,10 @@ helpers.jsx = () => template.program.ast`
 helpers.asyncIterator = () => template.program.ast`
   export default function _asyncIterator(iterable) {
     if (typeof Symbol === "function") {
-      if (Symbol.asyncIterator) {
-        var method = iterable[Symbol.asyncIterator];
-        if (method != null) return method.call(iterable);
+      if (Symbol.asyncIterator && iterable[Symbol.asyncIterator] != null) {
+        return iterable[Symbol.asyncIterator]();
```

Comment:
After reading this again, it has a little bug: it gets `iterable[Symbol.asyncIterator]` and `iterable[Symbol.iterator]` twice.
It should be something like this:
```js
if (Symbol.asyncIterator && (method = iterable[Symbol.asyncIterator]) != null) return method.call(iterable);
```

Reply:
whats wrong with accessing it twice?

Reply:
```js
var i = 0;

var obj = {
  get [Symbol.asyncIterator]() {
    i++;
    return async function* () { yield 1; yield 2; };
  }
};

(async () => {
  for await (const x of obj) {}
  alert(i); // Should alert 1, not 2
})();
```

Reply:
^ Actually this should be a test, but it can be added in a different PR.

Reply:
that seems like a bit of a contrived example, but sure

Reply:
Usually if it doesn't add too much complexity and it isn't slower, we try to be as much spec compliant as possible.